jetty-ee9/jetty-ee9-nested/src/test/java/org/eclipse/jetty/ee9/nested/NcsaRequestLogTest.java

```diff
@@ -161,12 +185,13 @@ public void testAbsolute(String logType) throws Exception
         setup(logType);
         testHandlerServerStart();
 
-        _connector.getResponse(
-            "GET http://hostname:8888/foo?name=value HTTP/1.1\n" +
-                "Host: hostname:8888\n" +
-                "\n");
+        _connector.getResponse("""
+            GET http://hostname:8888/foo?name=value HTTP/1.1
+            Host: hostname:8888
+            
+            """);
         String log = _entries.poll(5, TimeUnit.SECONDS);
-        assertThat(log, containsString("GET http://hostname:8888/foo?name=value"));
+        assertThat(log, containsString("GET /foo?name=value"));
```

Comment:
We don't have access in Jetty 12 to get the URI as it was received, so the request log reconstructs it using `HttpURI.getPathQuery()`.

I think to fix this we would need some new API on `Request`.

Reply:
The "URI as it was received" is a strange one.
We build the URI from the components on the HTTP protocol (host, path, and query), and information from the connector (like isSecure).
What we see down here is the URI after it has possibly been modified (like the RewriteHandler), right?
If we had the raw host/path as seen on the protocol, would that be sufficient? or would that possibly cause NCSA format/syntax issues? (like what if the path was received on HTTP/2 with a space or control character that we stripped out / changed due to a Rewrite Rule?)

Reply:
The NCSA combined format can be represented like this ...

```
%h %l %u %t "%r" %s %b "%{Referer}i" "%{User-agent}i"

    %h: the client hostname or IP address
    %l: the client identifier or - if none is set
    %u: the username used by the client for authentication or - if none is set
    %t: the date and time of the HTTP request
    %r: the first line of the HTTP request
    %s: the status code of the HTTP response
    %b: the byte size of the HTTP response
    %{Referer}i: the URL that linked to the requested page or - if none is set
    %{User-agent}i: the web browser and platform used by the client
```

I think you are attempting to handle the `%r` with this, right?
If so, this would really only work for HTTP/1.x, not HTTP/2 or HTTP/3 (as those two don't have the concept of "the first line of the HTTP request")

This "first line of the HTTP request" could also help with your `BAD /badRequest HTTP/1.0` issue below.

Reply:
I worry that if we implement this, someone nefarious sort could come along and just hammer a server with huge "first line" and blow out the request log and/or server memory (as we would now store that raw "first line" in a way accessible from the `Request` object).

Reply:
I think its probably fine to keep the behavior for this as is. Even though it might not be the exact first line of the request for HTTP/1.x.

Reply:
I'm fine with this PR being what it is.
I opened #12423 to cover the other things brought up though.